src/utils.ts

```diff
@@ -137,15 +208,23 @@ export function parseSkylink(skylinkStr: string, opts: any = {}): string {
   const matchPathname = skylinkAndPath.match(SKYLINK_PATHNAME_REGEX);
   if (!matchPathname) return null;
 
-  let path = matchPathname[SKYLINK_PATH_MATCH_POSITION];
-  if (path == "/") path = "";
+  const path = matchPathname[SKYLINK_PATH_MATCH_POSITION];
 
   if (opts.includePath) return trimForwardSlash(skylinkAndPath);
   else if (opts.onlyPath) return path;
   else return matchPathname[SKYLINK_DIRECT_MATCH_POSITION];
 }
 
-function parseSkylinkBase32(skylinkStr: string, opts: any = {}): string {
+/**
+ * Parses the given string for a base32 skylink.
+ *
+ * @param skylinkStr - Base32 skylink.
+ * @param [opts] - Additional settings that can optionally be set.
+ * @returns - The base32 skylink.
+ */
+export function parseSkylinkBase32(skylinkStr: string, opts?: ParseSkylinkBase32Options): string | null {
+  opts = { ...opts };
```

Comment:
Does this basically ensure that `opts` is not null?

Reply:
Right. I could have given `opts` a default value of `{}` instead, but I wanted to be consistent with how we fill in the options in methods like `uploadFile` and `downloadFile`. Maybe a default value is more clear though. I think I'll change it since `opts = { ...opts }` does look kind of weird.